### PR TITLE
Feature/use grid layout

### PR DESCRIPTION
**Pull request by mr-mig**
@dev-tim You should be unblocked after the merge

### CODE REVIEW

File: public/app/app.jsx [hunk 1]
```diff
@@ -17,26 +17,15 @@ var NotFoundRoute = Router.NotFoundRoute;
...
-      <Route name="settings" handler={SettingsRegion}/>
-      <Route name="devices" handler={DevicesRegion}/>
-      <Redirect from="/" to="devices"/>
-    </Route>
-    <Redirect from="/" to="main"/>
+  <Route path="/" handler={MainScreen}>
+    <Route name="settings" handler={SettingsRegion}/>
+    <Route name="devices" handler={DevicesRegion}/>
+    <Redirect from="/" to="devices"/>
   </Route>
```
Comment (dev-tim) [inline public/app/app.jsx]: You decided to get rid of subrouting for now?

Comment (larsblumberg) [inline public/app/app.jsx]: Yes, there's currently no need for it. Can be added if necessary

File: public/app/screens/main/main-screen.jsx [hunk 2]
```diff
@@ -3,35 +3,40 @@ var React = require("react");
...
+      navItems: navigationStore.getItems()
+    }
+  },
+
+  render: function () {
+    var navigation = <Navigation items={this.state.navItems} orientation="medium-vertical small-horizontal"/>;
+
+    return (
+      <div className="main-screen">
+        <div className="main-screen__top-row main-screen--small-only">
```
Comment (larsblumberg) [inline public/app/screens/main/main-screen.jsx]: As we have `main-screen--hidden-for-small` this could be named `main-screen--visible-for-small-only` to have a naming pendant.

File: public/app/screens/main/navigation/navigation.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+var Navigation = React.createClass({
+  mixins: [StyleMixin(require("./style.scss"))],
+  render: function () {
+    // TODO: Use mixins instead of CSS classes
+    // TODO: Pass number of elements to mixin instead of specifying class name
+    var classes = "cc_navigation icon-bar two-up " + this.props.orientation;
+    
+    var items = this.props.items.map(function (item, i) {
+      return (
+        <Link to={item.linkTo} key={i} className="cc_navigation__link item">
```
Comment (larsblumberg) [inline public/app/screens/main/navigation/navigation.jsx]: class `item` should be included by mixin class `cc_navigation__link`?

File: public/app/screens/main/navigation/navigation.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+var React = require('react'),
+  Link = require('react-router').Link,
+  StyleMixin = require('mixins/style-mixin');
+
+var Navigation = React.createClass({
+  mixins: [StyleMixin(require("./style.scss"))],
+  render: function () {
+    // TODO: Use mixins instead of CSS classes
+    // TODO: Pass number of elements to mixin instead of specifying class name
+    var classes = "cc_navigation icon-bar two-up " + this.props.orientation;
```
Comment (larsblumberg) [inline public/app/screens/main/navigation/navigation.jsx]: Classes `icon-bar two-up` should be included by `cc_navigation`? This could however by done by a different PR that concentrates on using mixins for this icon-bar

File: public/app/project-setup/theme/variables/_base.scss [hunk 5]
```diff
@@ -6,4 +6,10 @@ $include-html-global-classes: true;
 
 // this is used for Headers
 $include-html-type-classes: true;
-$include-html-grid-classes: true;
+
+$include-html-grid-classes: false;
+
+// this is used as extensions for sass-based styles
+$include-html-visibility-classes: true;
+
```
Comment (dev-tim) [inline public/app/project-setup/theme/variables/_base.scss]: :+1:

File: public/app/screens/main/navigation/style.scss [hunk 6]
```diff
@@ -0,0 +1,12 @@
+@import "theme";
+
+.cc_navigation {
+  @include icon-bar();
+
+  .cc_navigation__link {
+    // .active is applied by foundation automatically
```
Comment (larsblumberg) [inline public/app/screens/main/navigation/style.scss]: Comment could be more descriptive:
`// .active is applied to selected link automatically by foundation`

File: public/app/project-setup/theme/variables/_grid.scss [hunk 7]
```diff
@@ -1,5 +1,5 @@
 @import "foundation/functions";
 
-$row-width: rem-calc(1000);
+$row-width: 1000px;
```
Comment (dev-tim) [inline public/app/project-setup/theme/variables/_grid.scss]: 1000px is a requirement from design guys?

File: public/app/screens/main/navigation/navigation.jsx (hunk 3 above)
Comment (dev-tim) [inline public/app/screens/main/navigation/navigation.jsx]: Do we need key here?

File: public/app/screens/main/sections/devices/device/device.jsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+var React = require('react');
+var StyleMixin = require('mixins/style-mixin');
+
+var Device = React.createClass({
+  mixins: [StyleMixin(require('./style.scss'))],
+  render: function () {
+    return (
+      <div className="cc-devices__device row">
+        <div className="cc-devices__device--title small-6 column">{this.props.device.title}</div>
```
Comment (dev-tim) [inline public/app/screens/main/sections/devices/device/device.jsx]: Shouldn't we move small-6 to mixin?

**Comment by larsblumberg** [comment]
LGTM, there are however some comments

File: public/app/project-setup/theme/variables/_grid.scss (hunk 7 above)
Comment (larsblumberg) [inline public/app/project-setup/theme/variables/_grid.scss]: yes

File: public/app/app.jsx (hunk 1 above)
Comment (dev-tim) [inline public/app/app.jsx]: Yep, anyway we do know how it works.

**Comment by dev-tim** [comment]
Me as well

File: public/app/screens/main/navigation/navigation.jsx (hunk 3 above)
Comment (larsblumberg) [inline public/app/screens/main/navigation/navigation.jsx]: Yes. The answer is the same as to the same question you did recently. It's needed by react: https://coderwall.com/p/jdybeq/the-importance-of-component-keys-in-react-js

File: public/app/screens/main/sections/devices/room/room.jsx [hunk 9]
```diff
@@ -1,7 +1,9 @@
 'use strict';
 
 var React = require('react');
-var Device = require('screens/main/devices/device/index');
+var Device = require('../device');
+
+require('./style.scss');
```
Comment (DaQuirm) [inline public/app/screens/main/sections/devices/room/room.jsx]: Why StyleMixin isn't used here?

Comment (larsblumberg) [inline public/app/screens/main/sections/devices/room/room.jsx]: was not part of the PR, we worked on the main screen and navigation bar. This could be done by a separate PR

File: public/app/screens/main/navigation/navigation.jsx (hunk 3 above)
Comment (mr-mig) [inline public/app/screens/main/navigation/navigation.jsx]: @larsblumberg two comments:
1. `item` should be fixed in a separate scope, as it is not a part of the template
2. `key` is not needed, if you are ok with the array indices used as keys. React do that automatically (https://github.com/facebook/react/issues/1342#issuecomment-39274004)

File: public/app/screens/main/sections/devices/device/device.jsx (hunk 8 above)
Comment (mr-mig) [inline public/app/screens/main/sections/devices/device/device.jsx]: This is NOT a part of template scope - it should be addressed when dealing with `device` component

File: public/app/screens/main/navigation/navigation.jsx (hunk 3 above)
Comment (larsblumberg) [inline public/app/screens/main/navigation/navigation.jsx]: Not defining the `key` property will however fire a warning onto the console. This is for good reason as the loop variable might not always be a good key (see the blog entry). This said I would prefer to always define a `key` explicitly so we never forget it.
